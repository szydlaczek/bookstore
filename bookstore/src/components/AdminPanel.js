import React from 'react';
import LoginPanel from './LoginPanel';
import BookForm from './BookForm';

class AdminPanel extends React.Component {
    constructor(){
        super();
        this.state= {
            
            loggedIn : false,    

        }

    }
    

    





changeLoggedIn = (newValue) => this.setState({loggedIn: newValue })

render() { ///musi byc zawsze
        
        return (
          
         <div>
             {!this.state.loggedIn &&
                <LoginPanel changeLoggedIn={this.changeLoggedIn}/>
             }
             {this.state.loggedIn &&
                <BookForm/>
             }
        </div>
        );
    }
}

export default AdminPanel;