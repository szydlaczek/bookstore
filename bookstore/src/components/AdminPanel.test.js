import React from 'react';
import ReactDOM from 'react-dom';
import AdminPanel from './AdminPanel'
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { wrap } from 'module';

configure({adapter: new Adapter()});

describe('App tests', () => {
    
    it('App renders without a problem', ()=> {
        const div=document.createElement('div');
        ReactDOM.render(<AdminPanel />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('App renders', () => {
        const wrapper=shallow(<AdminPanel/>);
        // console.log(wrapper.debug());
        expect(wrapper).toMatchSnapshot();
    })
})