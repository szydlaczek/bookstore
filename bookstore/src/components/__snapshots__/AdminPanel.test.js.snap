// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests App renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="name"
            name="name"
            placeholder="Book name"
            type="text"
            value="nodejs"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="author"
            name="author"
            placeholder="Book author"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <textarea
            className="form-control"
            id="description"
            name="description"
            placeholder="Book description"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-check-input"
            id="onStock"
            name="oonStock"
            type="checkbox"
          />
          <label
            className="form-check-label"
            htmlFor="bookOnStock"
          >
            On stock
          </label>
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="image"
            name="image"
            placeholder="Book image"
            type="text"
          />
        </div>
        <button
          className="btn btn-primary"
          type="submit"
        >
          Add
        </button>
      </form>,
      "className": "col-lg-4 adminPanel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="name"
              name="name"
              placeholder="Book name"
              type="text"
              value="nodejs"
            />
          </div>,
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="author"
              name="author"
              placeholder="Book author"
              type="text"
            />
          </div>,
          <div
            className="form-group"
          >
            <textarea
              className="form-control"
              id="description"
              name="description"
              placeholder="Book description"
            />
          </div>,
          <div
            className="form-group"
          >
            <input
              className="form-check-input"
              id="onStock"
              name="oonStock"
              type="checkbox"
            />
            <label
              className="form-check-label"
              htmlFor="bookOnStock"
            >
              On stock
            </label>
          </div>,
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="image"
              name="image"
              placeholder="Book image"
              type="text"
            />
          </div>,
          <button
            className="btn btn-primary"
            type="submit"
          >
            Add
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              id="name"
              name="name"
              placeholder="Book name"
              type="text"
              value="nodejs"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "name",
              "name": "name",
              "placeholder": "Book name",
              "type": "text",
              "value": "nodejs",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              id="author"
              name="author"
              placeholder="Book author"
              type="text"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "author",
              "name": "author",
              "placeholder": "Book author",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <textarea
              className="form-control"
              id="description"
              name="description"
              placeholder="Book description"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "description",
              "name": "description",
              "placeholder": "Book description",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form-check-input"
                id="onStock"
                name="oonStock"
                type="checkbox"
              />,
              <label
                className="form-check-label"
                htmlFor="bookOnStock"
              >
                On stock
              </label>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-check-input",
                "id": "onStock",
                "name": "oonStock",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "On stock",
                "className": "form-check-label",
                "htmlFor": "bookOnStock",
              },
              "ref": null,
              "rendered": "On stock",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              id="image"
              name="image"
              placeholder="Book image"
              type="text"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "image",
              "name": "image",
              "placeholder": "Book image",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "className": "btn btn-primary",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="name"
              name="name"
              placeholder="Book name"
              type="text"
              value="nodejs"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="author"
              name="author"
              placeholder="Book author"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <textarea
              className="form-control"
              id="description"
              name="description"
              placeholder="Book description"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-check-input"
              id="onStock"
              name="oonStock"
              type="checkbox"
            />
            <label
              className="form-check-label"
              htmlFor="bookOnStock"
            >
              On stock
            </label>
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="image"
              name="image"
              placeholder="Book image"
              type="text"
            />
          </div>
          <button
            className="btn btn-primary"
            type="submit"
          >
            Add
          </button>
        </form>,
        "className": "col-lg-4 adminPanel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="name"
                name="name"
                placeholder="Book name"
                type="text"
                value="nodejs"
              />
            </div>,
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="author"
                name="author"
                placeholder="Book author"
                type="text"
              />
            </div>,
            <div
              className="form-group"
            >
              <textarea
                className="form-control"
                id="description"
                name="description"
                placeholder="Book description"
              />
            </div>,
            <div
              className="form-group"
            >
              <input
                className="form-check-input"
                id="onStock"
                name="oonStock"
                type="checkbox"
              />
              <label
                className="form-check-label"
                htmlFor="bookOnStock"
              >
                On stock
              </label>
            </div>,
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="image"
                name="image"
                placeholder="Book image"
                type="text"
              />
            </div>,
            <button
              className="btn btn-primary"
              type="submit"
            >
              Add
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                id="name"
                name="name"
                placeholder="Book name"
                type="text"
                value="nodejs"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "name",
                "name": "name",
                "placeholder": "Book name",
                "type": "text",
                "value": "nodejs",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                id="author"
                name="author"
                placeholder="Book author"
                type="text"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "author",
                "name": "author",
                "placeholder": "Book author",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <textarea
                className="form-control"
                id="description"
                name="description"
                placeholder="Book description"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "description",
                "name": "description",
                "placeholder": "Book description",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="form-check-input"
                  id="onStock"
                  name="oonStock"
                  type="checkbox"
                />,
                <label
                  className="form-check-label"
                  htmlFor="bookOnStock"
                >
                  On stock
                </label>,
              ],
              "className": "form-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-check-input",
                  "id": "onStock",
                  "name": "oonStock",
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "On stock",
                  "className": "form-check-label",
                  "htmlFor": "bookOnStock",
                },
                "ref": null,
                "rendered": "On stock",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                id="image"
                name="image"
                placeholder="Book image"
                type="text"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "image",
                "name": "image",
                "placeholder": "Book image",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
